### DISCUSSION TITLE
Possible to just capture and save users credit card ?

### DISCUSSION THREAD

**Discussion by SogoGolf** [flutter-stripe/flutter_stripe#1020] [Q&A]
We have a requirement from our Flutter app, where the user should be able to enter their payment detalis (credit card), and have those details saved to Stripe so they can be used later (eg. end of each month).

How can this be achieved using this library ? Whilst we had no problems showing the credit card _payment_ popup provided by this library, we couldn't see any UI for just entering and saving credit card details. Is this possible to do?

**Comment by jonasbark** [comment]
Use setup intents
https://www.google.com/search?q=setup+intents+payment+sheet&oq=setup+intents+payment+sheet

**Comment by SogoGolf** [comment]
hey @jonasbark thanks for quick reply. i _think_ (not sure) there might be a problem with flutter_stripe but perhaps we are also not calling the API's in correct order. At the moment we keep getting an error ```Card detail is not complete```. Can you let me know if what we are doing is correct ?

1. user taps button in Flutter (iOS) app (to proceed with recording their credit card on Stripe for future purchases)

2. we call a simple nodejs server app that returns customer id, client secret, ephemeral key and publishableKey. Here is the nodejs server side code:

```
  const customer = await stripe.customers.retrieve("xxxxxxxxxxx");

  const ephemeralKey = await stripe.ephemeralKeys.create(
    { customer: customer.id },
    { apiVersion: "2020-08-27" }
  );

  const setupIntent = await stripe.setupIntents.create({
    customer: customer.id,
    payment_method_types: ["card"],
  });
  res.json({
    clientSecret: setupIntent.client_secret,
    ephemeralKey: ephemeralKey.secret,
    customer: customer.id,
    publishableKey:
      "pk_test_xxxxxxxxxxxx",
  });
```

**This works fine and we have a response payload from the server to use. Now, back in the Flutter app ->** 

3. call ```Stripe.instance.initPaymentSheet```. (Notice we are using a param called ```setupIntentClientSecret``` also, rather than ```paymentIntentClientSecret```)
```
await Stripe.instance.initPaymentSheet(
        paymentSheetParameters: SetupPaymentSheetParameters(
      customerId: serverResponse!['customer'],
      customerEphemeralKeySecret: serverResponse!['ephemeralKey'],
      setupIntentClientSecret: serverResponse!['clientSecret'],
      style: ThemeMode.dark,
      merchantDisplayName: 'Test',
      allowsDelayedPaymentMethods: true,
    ));
```

4. display the payment sheet so user can enter their credit card deets
```
await Stripe.instance.presentPaymentSheet();
```

The app displays Stripe's payment collection popup, looks like this:
<img width="166" alt="IMG_AE05D5793C59-1" src="https://user-images.githubusercontent.com/19580787/201785152-23147290-249b-4deb-b715-1070dc0522ad.png">


5. User taps on "Setup" button


6. now call ```Stripe.instance.confirmSetupIntent``` ; this is where i think we are wrong in how the APIs are being used ? 
```
final setupIntentResult = await Stripe.instance.confirmSetupIntent(
      paymentIntentClientSecret: serverResponse!['clientSecret'],
      params: const PaymentMethodParams.card(
        paymentMethodData: PaymentMethodData(
          billingDetails: BillingDetails(.....snip),
          shippingDetails: ShippingDetails(...snip),
          ),
        ),
      ),
    );
```

ERROR ! This is where the error occurs. Here is error:
<img width="429" alt="2022-11-15_09-52-32" src="https://user-images.githubusercontent.com/19580787/201785472-327590f5-52cd-416d-8b9f-13139793a9f5.png">

I still think we are not calling the API's correctly ? Or is this a bug with the flutter_stripe lib on iOS ?
